vmngclient/api/templates.py

```diff
@@ -0,0 +1,240 @@
+            else:
+                return True
+
+        @retry(
+            wait=wait_fixed(sleep_seconds),
+            stop=stop_after_attempt(int(timeout_seconds / sleep_seconds)),
+            retry=retry_if_result(check_status),
+        )
+        def wait_for_status():
+            return self.get_operation_status(operation_id)
+
+        return True if wait_for_status() else False
+
+    def attach(self, template_name: str, device: DeviceInfo) -> bool:
+        """Attach template to device.
+
+        Args:
+            template_name (str): Template name to attached.
+            device (DeviceInfo): Device to attach template.
+
+        Returns:
+            bool: True if attached template is successful, otherwise - False.
+        """
+        templateId = self.get_id(template_name)
+        payload = {
+            "deviceTemplateList": [
+                {
+                    "templateId": templateId,
+                    "device": [
+                        {
+                            "csv-status": "complete",
+                            "csv-deviceId": device.uuid,
+                            "csv-deviceIP": device.id,
+                            "csv-host-name": device.hostname,
+                            "csv-templateId": templateId,
+                        }
+                    ],
+                }
+            ]
+        }
+        endpoint = "/dataservice/template/device/config/attachcli"
+        response = cast(dict, self.session.post_json(url=endpoint, data=payload))
+        return self.wait_complete(response['id'])
+
+    def device_to_cli(self, device: DeviceInfo) -> bool:
+        """Device mode to Cli.
+
+        Args:
+            device (DeviceInfo): Device to chcange mode.
+
+        Returns:
+            bool: True if change mode to cli is successful, otherwise - False.
+        """
+        payload = {"deviceType": device.personality, "devices": [{"deviceId": device.uuid, "deviceIP": device.id}]}
+        endpoint = "/dataservice/template/config/device/mode/cli"
+        response = cast(dict, self.session.post_json(url=endpoint, data=payload))
+        return self.wait_complete(response['id'])
+
+    def get_operation_information(self, id: str) -> List[Dict[str, str]]:
+        """Get operation information.
+
+        Args:
+            id (str): Operatrion id.
+
+        Returns:
+            List[Dict[str, str]]: List action status.
+        """
+        endpoint = f"/dataservice/device/action/status/{id}"
+        response = cast(list, self.session.get_data(endpoint))
+        return response
+
+    def get_operation_status(self, operation_id: str) -> List[OperationStatus]:
+        """Get operatrion status.
+
+        Args:
+            operation_id (str): Operatrion id.
+
+        Returns:
+            List[OperationStatus]: List of operations.
+        """
+        return [OperationStatus(status['status']) for status in self.get_operation_information(operation_id)]
```

Comment:
Why not make OperationStatus a dataclass?
Is OparationStatus specific to template api operation? or is it general for all queries to /dataservice/device/action/status/?
In first case i would put it in the same file as api classes (this file)
In second case i would put it in dataclasses and name it DeviceActionStatus

Reply:
It only reads this one value - I don't see any point in creating a dataclass. Operation status - it is used, it has been used in several places, therefore it is in utils.